src/closure.jl

```diff
@@ -6,7 +6,7 @@ function getgensym!(defined::Dict{Symbol,Symbol}, s::Symbol)
   return snew
 end
 
-extractargs!(arguments::Vector{Symbol}, defined::Dict{Symbol,Symbol}, sym, mod) = nothing
+extractargs!(arguments::Vector{Symbol}, defined::Dict{Symbol,Symbol}, sym, mod; skipsymbol=:()) = nothing
```

Comment:
Note that `:()` is an `Expr`.
You can use `nothing` if you want an object of a different type, or something like `Symbol("")` or `Symbol("#reserved#polyester#thread#local#storage#")` if you'd like a symbol that can't be constructed with a literal normally.
Someone could of course write code writing `var""` or `var#reserved#polyester#thread#local#storage#`. But that won't happen in normal code.

I'd prefer a `Symbol` for the sake of latency, i.e. lets avoid compiling any more specializations here than we have to.